core/domain/feedback_services.py

```diff
@@ -139,81 +139,142 @@ def create_message(
+    for each message, subscribes the author to the thread.
+
+    Args:
+        thread_ids: list(str). The thread ids the messages belong to.
+        author_id: str. The author id who creates the messages.
+        updated_status: str. One of STATUS_CHOICES. New thread status.
+            Must be supplied if this is the first message of the threads.
+            Otherwise, this property should only exist when the status
+            changes.
+        updated_subject: str. New thread subject. Must be supplied if this is
+            the first message of the threads. Otherwise, this property should
+            only exist when the subject changes.
+        text: str. The text of the feedback message. This may be ''.
+        received_via_email: bool. Whether new the message(s) are received via
+            email or web.
+    """
     from core.domain import event_services
-    # Get the thread at the outset, in order to check that the thread_id passed
-    # in is valid.
-    thread = feedback_models.GeneralFeedbackThreadModel.get(thread_id)
-
-    message_id = feedback_models.GeneralFeedbackMessageModel.get_message_count(
-        thread_id)
-    message = feedback_models.GeneralFeedbackMessageModel.create(
-        thread_id, message_id)
-    message.thread_id = thread_id
-    message.message_id = message_id
-    message.author_id = author_id
-    message.text = text
-    message.received_via_email = received_via_email
-    if updated_status:
-        message.updated_status = updated_status
-        if message_id == 0:
-            # New thread.
-            if thread.entity_type == feconf.ENTITY_TYPE_EXPLORATION:
-                event_services.FeedbackThreadCreatedEventHandler.record(
-                    thread.entity_id)
-        else:
-            # Thread status changed.
-            if thread.entity_type == feconf.ENTITY_TYPE_EXPLORATION:
-                event_services.FeedbackThreadStatusChangedEventHandler.record(
-                    thread.entity_id, thread.status, updated_status)
-    if updated_subject:
-        message.updated_subject = updated_subject
-    message.put()
-
-    # Update the message data cache of the thread.
-    if text:
-        thread.last_nonempty_message_text = text
-        thread.last_nonempty_message_author_id = author_id
-    if thread.message_count is not None:
+
+    # Get the threads at the outset, in order to check that there are models
+    # corresponding to each of the thread_ids.
+    thread_models = feedback_models.GeneralFeedbackThreadModel.get_multi(
+        thread_ids)
+    thread_ids_that_do_not_have_models = []
+    for index, thread in enumerate(thread_models):
+        if thread is None:
+            thread_ids_that_do_not_have_models.append(thread_ids[index])
+    if len(thread_ids_that_do_not_have_models) > 0:
+        raise Exception(
+            'Thread(s) belonging to the GeneralFeedbackThreadModel class with '
+            'id(s):[%s] were not found.' % (
```

Comment:
It should be possible to pluralize this correctly -- do something like `'id%s:[%s] ...' % ('s' if len(...) > 1 else '', ' '.join(...))`